### PR TITLE
Feature/issue 1019 array gocb urls

### LINKED ISSUE #1019: Support an array of URLs for server key for startup handling
From #942 - we're using go-couchbase's bucketUpdater handling to support the rebalance/dropped connection handling.

However, we still need a way for Sync Gateway to try multiple URLs when initially making the bucket connection.  This can be implemented strictly on the Sync Gateway side.

### PR DESCRIPTION
**Pull request by tleyden**
Fixes #1019 to support an array of urls

For full SG-Accel support, there is still outstanding work: https://github.com/couchbaselabs/sync-gateway-accel/issues/163

- [x] [SG-Accel PR](https://github.com/couchbaselabs/sync-gateway-accel/pull/164)
- [x] Manifest update to point to latest SG-accel
- [x] [Unit-integration test passed](http://uberjenkins.sc.couchbase.com/view/Build/job/build-sync-gateway/162/)

### CODE REVIEW

File: base/util_test.go [hunk 1]
```diff
@@ -192,5 +191,67 @@ func TestHighSeqNosToSequenceClock(t *testing.T) {
 
 }
 
+func TestCouchbaseURIToHttpURL(t *testing.T) {
+
+	inputsAndExpected := []struct {
```
Comment (tleyden) [inline base/util_test.go:196]: TODO: add basic auth credentials to an http url and see if they are preserved

**Comment by tleyden** [comment]
@adamcfraser in 38302f469b71b56bb190302ecd0f4c1307dc0e11 I extended that unit test to include basic auth strings, but it didn't pass:

```
	util_test.go:260
	assert.DeepEquals(t, actual, inputAndExpected.expected)	

	expected: []string{"http://foo:bar@host1:8091"}
	     got: []string{"http://host1:8091"}
	assert.go:31: 

```

Looking into a fix

**Comment by tleyden** [comment]
@adamcfraser I pushed a commit that handles backwards compatibility, and the unit tests pass now.  I was wrong about the "couchbase://" scheme error, I swear I saw that at one point.  Also, surprisingly, the url parser parses strings like "http://host1,host2", and so I added a check for hosts with a comma.

**Comment by valeriomazzeo** [comment]
in which version of sync-gateway is this going to be available?

**Comment by tleyden** [comment]
1.5

**Pull request merged**